src/Controller/FilesController.php

```diff
@@ -78,16 +81,28 @@ public function fileAction(Request $request, string $file): BinaryFileResponse
         $allowLogin = false;
         $allowAccess = false;
 
+        // Get the current user
+        $user = $this->security->getUser();
+
+        // Check if the current user can access their home directory
+        $canAccessHomeDir = $user instanceof FrontendUser && !empty($user->homeDir) && $user->accessHomeDir;
+
         do {
             // Only check for folders and when member groups have been set
-            if ('folder' === $filesModel->type && null !== $filesModel->groups) {
+            // or access to member home directory
+            if ('folder' === $filesModel->type
+                && (null !== $filesModel->groups
+                    || ($canAccessHomeDir))) {
                 $allowLogin = true;
 
                 // Set the model to protected on the fly
                 $filesModel->protected = true;
 
+                // Check if this file is the home directory
+                $isHomeDir = $user->homeDir === $filesModel->uuid;
```

Comment:
```suggestion
                $isHomeDir = $user instanceof FrontendUser && $user->homeDir === $filesModel->uuid;
```

`$user` might be `null` or not a `FrontendUser` instance.